spec.emu

```diff
@@ -268,93 +272,231 @@ contributors: Ujjwal Sharma, Younies Mahmoud
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-partitiondurationformatpattern" type="abstract operation">
+    <emu-clause id="sec-setfractionaldigitoptions" type="abstract operation">
       <h1>
-        PartitionDurationFormatPattern (
-          _durationFormat_: a DurationFormat,
+        SetFractionalDigitOptions (
+        _durationFormat_: a DurationFormat Object,
+        _nfOpts_: An Object,
+        ): ~unused~
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It creates properties on _nfOpts_ needed for formatting fractional units.</dd>
+      </dl>
+
+      <emu-alg>
+        1. If _durationFormat_.[[FractionalDigits]] is *undefined*, then
+          1. Let _maximumFractionDigits_ be *9*<sub>𝔽</sub>.
+          1. Let _minimumFractionDigits_ be *+0*<sub>𝔽</sub>.
+        1. Else,
+          1. Let _maximumFractionDigits_ be _durationFormat_.[[FractionalDigits]].
+          1. Let _minimumFractionDigits_ be _durationFormat_.[[FractionalDigits]].
+        1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"maximumFractionDigits"*, _maximumFractionDigits_).
+        1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"minimumFractionDigits"*, _minimumFractionDigits_).
+        1. Perform ! CreateDataPropertyOrThrow(_nfOpts_, *"roundingMode"*, *"trunc"*).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-nextunitfractional" type="abstract operation">
+      <h1>
+        NextUnitFractional (
+          _durationFormat_: a DurationFormat Object,
           _duration_: a Duration Record,
-        ): a List
+          _unit_: a String,
+        ): a Boolean
       </h1>
       <dl class="header">
         <dt>description</dt>
-        <dd>It creates and returns a List with all the corresponding parts according to the effective locale and the formatting options of _durationFormat_.</dd>
+        <dd>It returns *true* if the next smallest unit uses the *"fractional"* style.</dd>
+      </dl>
+
+      <emu-alg>
+        1. Assert: _unit_ is *"seconds"*, *"milliseconds"*, or *"microseconds"*.
+        1. If _unit_ is *"seconds"*, then
+          1. If _durationFormat_.[[MillisecondsStyle]] is *"fractional"*, return *true*.
+        1. If _unit_ is *"milliseconds"*, then
+          1. If _durationFormat_.[[MicrosecondsStyle]] is *"fractional"*, return *true*.
+        1. If _unit_ is *"microseconds"*, then
+          1. If _durationFormat_.[[NanosecondsStyle]] is *"fractional"*, return *true*.
+        1. Return *false*.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-formathours" type="abstract operation">
+      <h1>
+        FormatHours (
+          _durationFormat_: a DurationFormat Object,
```

Comment:
how about we only pass in  _durationFormat_.[[Locale]] and _durationFormat_.[[HoursStyle]] but not passing in _durationFormat_?